library/hatomap.py

```diff
@@ -0,0 +1,478 @@
+                        color=self.fill_color,
+                        lineType=cv2.LINE_AA,
+                    )
+                cv2.rectangle(
+                    img,
+                    tl,
+                    br,
+                    thickness=self.border_width,
+                    color=self.border_color,
+                    lineType=cv2.LINE_AA,
+                )
+        elif self.symbol in symbols.keys():
+            symbol_coords = symbols[self.symbol]
+            px_coords = (
+                np.repeat(px_coords[:, np.newaxis, :], symbol_coords.shape[0], axis=1)
+                + symbol_coords * self.size
+            )
+            px_coords = px_coords.astype(np.int32)
+            for c in px_coords:
+                if self.fill_color is not None:
+                    cv2.fillPoly(img, [c], color=self.fill_color, lineType=cv2.LINE_AA)
+                cv2.polylines(
+                    img,
+                    [c],
+                    color=self.border_color,
+                    thickness=self.border_width,
+                    isClosed=True,
+                    lineType=cv2.LINE_AA,
+                )
+
+        return img
+
+
+@dataclass
+class RasterLayer:
+    url: Union[List[str], str]
+    opacity: float = 1.0
+    brightness: float = 1.0
+    chroma: float = 1.0
+
+    def __post_init__(self):
+        if not (0 <= self.opacity <= 1.0):
+            raise ValueError("Opacity is out of range.")
+
+    def get_image(self, bbox: WebMercatorPixelBBox = None) -> np.ndarray:
+        if type(self.url) is list:
+            url = random.choice(self.url)
+        else:
+            url = self.url
+        layer_img = RasterTileServer(url).request(bbox)
+
+        if self.brightness != 1.0 or self.chroma != 1.0:
+            img_hsv = cv2.cvtColor(layer_img, cv2.COLOR_BGR2HSV)
+            img_hsv[..., 1] = img_hsv[..., 1] * self.brightness
+            img_hsv[..., 2] = img_hsv[..., 2] * self.chroma
+            layer_img = cv2.cvtColor(img_hsv, cv2.COLOR_HSV2BGR)
+
+        if self.opacity != 1.0:
+            layer_img[layer_img[..., 3] != 0, 3] = int(round(self.opacity * 256))
+
+        return layer_img
+
+
+def cv2_putText_3(img, text, org, fontFace, fontScale, color):
+    """
+    OpenCVでも日本語フォントでテキストを挿入する。
+    # https://qiita.com/mo256man/items/b6e17b5a66d1ea13b5e3
+    """
+    x, y = org
+    imgPIL = Image.fromarray(img)
+    draw = ImageDraw.Draw(imgPIL)
+    fontPIL = ImageFont.truetype(font=fontFace, size=fontScale)
+    w, h = draw.textsize(text, font=fontPIL)
+    draw.text(xy=(x, y - h), text=text, fill=color, font=fontPIL)
+    return np.array(imgPIL, dtype=np.uint8)
+
+
+@dataclass
+class HatoMap:
+    """
+    mapbox: 地図の表示範囲と地図の大きさ
+    basemap: 描画する地図の種類（open-street-mapなど）
+    extra_basemap_server: 描画する地図をプリセット以外にする
+    layers: 地図の上から地図タイルや線・点などを描画する
+    title: タイトル
+    """
+
+    mapbox: MapBox = MapBox()
+    basemap: str = "open-street-map"
+    extra_basemap_server: str = None
```

Comment:
```suggestion
    extra_basemap_server: Optional[str] = None
```

https://github.com/dev-hato/hato-bot/actions/runs/2783525807/jobs/4380588635

```
library/hatomap.py:395: error: Incompatible types in assignment (expression has type "None", variable has type "str")
```